spec/test/collections/test_silva.py

```diff
@@ -0,0 +1,267 @@
+                    'name': 'name',
+                    'rank': 'fictional_rank',
+                },
+                "'fictional_rank' is not one of ['superfamily', 'subphylum', 'subfamily', " + \
+                "'phylum', 'order', 'major_clade', 'infraclass', 'suborder', 'family', " + \
+                "'superkingdom', 'domain', 'superphylum', 'superorder', 'superclass', " + \
+                "'infraphylum', 'subclass', 'genus', 'class', 'kingdom', 'subkingdom', " + \
+                "'root_rank', 'sequence']",
+            ), (
+                {
+                    # enum
+                    'id': 'id',
+                    'name': 'name',
+                    'rank': 'superclass',
+                    'dataset': ['nr99', 'ref', 'parc'],  # array in wrong order
+                },
+                "['nr99', 'ref', 'parc'] is not one of [['parc'], ['parc', 'ref'], ['parc', 'ref', 'nr99']]",
+            )
+        ]
+
+        cls.edges_valid = [
+            {
+                'id': '2',
+                'from': '2',
+                'to': '0',
+            }, {
+                'id': '42919',
+                'from': '42919',
+                'to': '2',
+            }, {
+                'id': 'HM032797.1.1344',
+                'from': 'HM032797.1.1344',
+                'to': '44300',
+            }, {
+                'id': 'CRQV01000019.5091.6588',
+                'from': 'CRQV01000019.5091.6588',
+                'to': '1853',
+            },
+        ]
+
+        cls.edges_invalid_errors = [
+            (
+                {
+                    # missing
+                    'from': '2',
+                    'to': '0',
+                },
+                "'id' is a required property",
+            ), (
+                {
+                    # missing
+                    'id': '2',
+                    'to': '0',
+                },
+                "'from' is a required property",
+            ), (
+                {
+                    # missing
+                    'id': '2',
+                    'from': '2',
+                },
+                "'to' is a required property",
+            ), (
+                {
+                    # type
+                    'id': 2,
+                    'from': '2',
+                    'to': '0',
+                },
+                "2 is not of type 'string'",
+            ), (
+                {
+                    # type
+                    'id': '2',
+                    'from': 2,
+                    'to': '0',
+                },
+                "2 is not of type 'string'",
+            ), (
+                {
+                    # type
+                    'id': '2',
+                    'from': '2',
+                    'to': 0,
+                },
+                "0 is not of type 'string'",
+            )
+        ]
+
+    def _test_type(self, schema, valid, invalid_errors):
+        for inst in valid:
+            run_validator(schema=schema, data=inst)
```

Comment:
you can add a `with self.subTest()` in here too

Reply:
thanks